streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1477,6 +1533,29 @@ public Object getStateLock() {
         return stateLock;
     }
 
+    public Map<String, KafkaFuture<Uuid>> consumerClientInstanceIds(final Duration timeout) {
+        final Map<String, KafkaFuture<Uuid>> result = new HashMap<>();
+
+        synchronized (fetchDeadlines) {
+            boolean addDeadline = false;
```

Comment:
Why do we need this boolean?

Reply:
This PR itself does not need it yet (but we need it later when we add support for restore consumer and producer)